cocotb/regression.py

```diff
@@ -305,23 +305,32 @@ class TestFactory(object):
     the test description) includes the name and description of each generator.
     """
 
-    def __init__(self, test_function, *args):
+    def __init__(self, test_function, name_postfix="", *args, **kwargs):
```

Comment:
I think this might cause problems.  I'm not convinced Python will like having a named kwarg before `*args` and then a `**kwargs`.  I forget which version but previously I've had to do something like this:

```
def my_func(*args, **kwargs):
    name_postfix = kwargs.get("name_postfix", None)
```

See also [this StackOverflow question](http://stackoverflow.com/questions/14003939/python-function-args-and-kwargs-with-other-specified-keyword-arguments)

I like the idea of passing `**kwargs` to the test function.  I suggest we make `postfix` (and maybe `prefix`) keyword arguments to `generate_tests()` instead, thus avoiding any confusion.

Would that work for your use-case?